elastic/tests/test_elastic.py

```diff
@@ -320,304 +345,318 @@
+def spin_up_elastic():
+    env = os.environ
+    args = [
+        'docker-compose', '-f', os.path.join(HERE, 'compose', 'elastic.yaml')
+    ]
+    subprocess.check_call(args + ["up", "-d"], env=env)
+    sys.stderr.write("Waiting for ES to boot...")
 
-        config = {
-            'instances': [
-                {'url': url, 'tags': tags},  # One with tags not external
-                {'url': url, 'cluster_stats': True},  # One without tags, external
-                {'url': bad_url},  # One bad url
-            ]
-        }
+    attempts = 1
 
-        self.assertRaises(
-            requests.exceptions.ConnectionError,
-            self.run_check, config=config, agent_config=agent_config)
+    while True:
+        if attempts >= 30:
+            subprocess.check_call(args + ["down"], env=env)
+            raise Exception("ES failed to boot...")
+        try:
+            res = requests.get(URL)
+            res.raise_for_status
+            break
+        except Exception:
+            attempts += 1
+            time.sleep(1)
 
-        default_tags = ["url:http://localhost:{0}".format(port)]
+    # Create an index in ES
+    requests.put(URL, '/datadog/')
+    yield
+    subprocess.check_call(args + ["down"], env=env)
 
-        expected_metrics = dict(STATS_METRICS)
-        CLUSTER_HEALTH_METRICS.update(CLUSTER_PENDING_TASKS)
-        expected_metrics.update(CLUSTER_HEALTH_METRICS)
 
-        instance_config = self.check.get_instance_config(config['instances'][0])
-        es_version = self.check._get_es_version(instance_config)
+@pytest.fixture
+def aggregator():
+    from datadog_checks.stubs import aggregator
+    aggregator.reset()
+    return aggregator
 
-        self.assertEquals(es_version, get_es_version())
 
-        if es_version >= [0, 90, 5]:
-            expected_metrics.update(ADDITIONAL_METRICS_POST_0_90_5)
-            if es_version >= [0, 90, 10]:
-                expected_metrics.update(JVM_METRICS_POST_0_90_10)
-            else:
-                expected_metrics.update(JVM_METRICS_PRE_0_90_10)
+def get_es_version():
+    version = os.environ.get("ELASTIC_VERSION")
+    dd_versions = {'0_90': [0, 90, 13], '1_0': [1, 0, 3], '1_1': [1, 1, 2], '1_2': [1, 2, 4]}
+    if version is None:
+        return [6, 0, 1]
+    if '_' in version:
+        return dd_versions[version]
+    else:
+        return [int(k) for k in version.split(".")]
+
+
+def test_bad_port(aggregator, spin_up_elastic):
+    elastic_check = ESCheck(CHECK_NAME, {}, {})
+    BAD_CONFIG = INSTANCE_CONFIG["instances"][1]
+    with pytest.raises(Exception):
+        elastic_check.check(BAD_CONFIG)
+
+
+def test_check(aggregator, spin_up_elastic):
+    elastic_check = ESCheck(CHECK_NAME, {}, {})
+    default_tags = ["url:http://{0}:{1}".format(HOST, PORT)]
+    CONFIG = INSTANCE_CONFIG["instances"][0]
+
+    attempts = 0
+    while True:
+        if attempts >= 3:
+            log.debug("Authentication Error occurred when running check")
```

Comment:
I added this loop because with the docker imaged provided by Elasticsearch, there is occasionally failures regarding authentication on the first check.

Reply:
👍 
I'd just add a small comment explaining that for clarity then 🙂